### ISSUE TITLE
Fitting twice with manual forecast lstm

### ISSUE THREAD

**Issue by srimantacse** [mikekeith52/scalecast#48]
I have created a forecaster like below

`f = Forecaster(
    y=df_data['y'],
    current_dates=df_data['ds'],
    test_length  = test_data_size,
    future_dates = test_data_size,
    cis = True,
    metrics = ['rmse','mape','mae','r2'],
)
alidation_split=.2,
    shuffle=True,
    activation='tanh',
    optimizer='Adam',
    learning_rate=0.0001,
    lstm_layer_sizes=(200,)*no_of_stack_lstm_layers,
    dropout=(0,)*no_of_stack_lstm_layers,
    callbacks=EarlyStopping(
      monitor='val_loss', 
      min_delta = 0.0001,
      patience=5
    ),
    plot_loss=True,
    verbose=True,
)`

@mikekeith52 
Why is it fitting twice?
Epoch 1/2
251/251 [==============================] - 10s 34ms/step - loss: 0.1126 - val_loss: 0.0990
Epoch 2/2
251/251 [==============================] - 9s 36ms/step - loss: 0.0957 - val_loss: 0.0981
1/1 [==============================] - 0s 465ms/step

Epoch 1/2
270/270 [==============================] - 21s 64ms/step - loss: 0.1123 - val_loss: 0.0942
Epoch 2/2
270/270 [==============================] - 13s 47ms/step - loss: 0.0962 - val_loss: 0.0937
1/1 [==============================] - 0s 363ms/step
85/85 [==============================] - 2s 29ms/step

**Comment by mikekeith52** [comment]
It is testing the model and then fitting it on the full dataset. To skip model testing and only fit once, where a forecast into a future horizon will still be generated, pass `test_length = 0` when initializing the `Forecaster` object.

**Comment by srimantacse** [comment]
My dataset shape is 
Train count:	2710
Test count:	192
batch size is 8, validation split is 0.2

So I am a bit confused, if the second iteration is for testing then how the iteration count is coming like 251/251 and 270/270 respectively as shown above?

**Comment by mikekeith52** [comment]
The first iteration tests the model using the training observations only. The second iteration fits the model on the full dataset. There are dropped observations to account for the size of the lags and the size of the forecast horizon. You can look over the source code: https://github.com/mikekeith52/scalecast/blob/main/src/scalecast/Forecaster.py#L748. To figure out why those exact numbers are being shown to you, I'd recommend digging into the TensorFlow docs, as the model is ported from that library.
